app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list) -> list[Person]:
+    spouse_types: set[str] = {"wife", "husband"}
+    for person in people:
+        if person["name"] in Person.people:
+            continue
+        current_person: Person = Person(person["name"], person["age"])
+        spouse_type: set = (spouse_types & person.keys())
+        if len(spouse_type) < 1 or person.get(next(iter(spouse_type))) is None:
```

Comment:
1. You can check `if not spouse_type` instead of len < 1
2. You don't need this check because from the solutioun you know there only will be valid dicts
3. Try to remove this next(iter()). You should care to make your code readable, it is hard to read now